src/trie/trie_type.c

```diff
@@ -231,12 +252,16 @@ void TrieType_RdbSave(RedisModuleIO *rdb, void *value) {
     rune *rstr;
     t_len len;
     float score;
+    RSPayload payload = {.data = NULL, .len = 0};
 
-    while (TrieIterator_Next(it, &rstr, &len, &score, NULL)) {
-      size_t slen;
+    while (TrieIterator_Next(it, &rstr, &len, &payload, &score, NULL)) {
+      size_t slen = 0;
       char *s = runesToStr(rstr, len, &slen);
       RedisModule_SaveStringBuffer(rdb, s, slen + 1);
       RedisModule_SaveDouble(rdb, (double)score);
+      // save an extra space for the null terminator to make the payload null terminated on load
+      if (payload.data != NULL && payload.len > 0)
```

Comment:
You are always reading the payload when loading from RDB, so in this case not saving palyloads will cause crashes.